lib/msf/core/opt_http_rhost_url.rb

```diff
@@ -51,7 +51,7 @@ def calculate_value(datastore)
       return unless datastore['RHOSTS']
       begin
         uri_type = datastore['SSL'] ? URI::HTTPS : URI::HTTP
-        uri = uri_type.build(host: datastore['RHOSTS'])
+        uri = uri_type.build(host: datastore['VHOST'] || datastore['RHOSTS'])
```

Comment:
if we have a `VHOST` set we should be attempting to rebuild the full url from that rather than just using the IP

Reply:
It'd be great to update the existing tests to cover this scenario :+1:

Reply:
hmmm I'll look into that, could be tricky though, the tests already _should_ be testing that, the problem here is something else is modifying it :grimacing:

Reply:
@adfoster-r7 sorted, I was wrong very easy after you helped me out :upside_down_face: